### PR TITLE
[NOD-1134] Integrate Protocol into main

### PR DESCRIPTION
**Pull request by svarogg**

### CODE REVIEW

File: netadapter/netadapter.go [hunk 1]
```diff
@@ -21,21 +21,31 @@ type NetAdapter struct {
...
 
 	return &adapter, nil
 }
 
+// Start begins the operation of the NetAdapter
+func (na *NetAdapter) Start() error {
+	return na.server.Start()
+}
+
+// Stop safely closes the netAdapter
```
Comment (someone235) [inline netadapter/netadapter.go:44]: netAdapter->NetAdapter

File: kaspad.go [hunk 2]
```diff
@@ -71,13 +81,19 @@ func newKaspad(listenAddrs []string, interrupt <-chan struct{}) (*kaspad, error)
 
 	txMempool := setupMempool(dag, sigCache)
 
+	protocolManager, err := protocol.NewProtocol(listenAddrs, dag)
```
Comment (stasatdaglabs) [inline kaspad.go:84]: I'd prefer if you just call it "protocol". If your linter is upset about it hiding the package, just call it p.

File: kaspad.go [hunk 3]
```diff
@@ -71,13 +81,19 @@ func newKaspad(listenAddrs []string, interrupt <-chan struct{}) (*kaspad, error)
 
 	txMempool := setupMempool(dag, sigCache)
 
+	protocolManager, err := protocol.New(listenAddrs, dag)
```
Comment (stasatdaglabs) [inline kaspad.go:84]: Please rename New to NewManager.

File: protocol/protocol.go [hunk 4]
```diff
@@ -6,25 +6,35 @@ import (
...
-type Protocol struct {
+// Manager manages the p2p protocol
+type Manager struct {
 	netAdapter *netadapter.NetAdapter
 }
 
-// Start starts the p2p protocol
-func Start(listeningPort string, dag *blockdag.BlockDAG) (*Protocol, error) {
-	netAdapter, err := netadapter.NewNetAdapter(listeningPort)
+// New creates a new instance of the p2p protocol
```
Comment (stasatdaglabs) [inline protocol/protocol.go:14]: This comment is now wrong. Please update it.

File: protocol/protocol.go [hunk 5]
```diff
@@ -6,25 +6,35 @@ import (
...
+	netAdapter, err := netadapter.NewNetAdapter(listeningAddrs)
 	if err != nil {
 		return nil, err
 	}
 
 	routerInitializer := newRouterInitializer(netAdapter, dag)
 	netAdapter.SetRouterInitializer(routerInitializer)
 
-	protocol := Protocol{
+	manager := Manager{
```
Comment (stasatdaglabs) [inline protocol/protocol.go:24]: Manager of what? Please call it protocolManager.

**Pull request merged**